File: 3227-find-missing-and-repeated-values/find-missing-and-repeated-values.py
class Solution:
    def findMissingAndRepeatedValues(self, grid: List[List[int]]) -> List[int]:
        ans = []
        res = []

        for i in grid:
            for j in i:
                ans.append(j)

        ast = list(set(ans))
        x = 0

        for i in ast:
            if ans.count(i) > 1:
                res.append(i)
        for i in range(1, len(ast)):
            if abs(ast[i - 1] - ast[i]) != 1:
                x = ast[i] - 1

        if x == 0 and ast[0] == 1:
            res.append(ast[len(ast) - 1] + 1)
        elif x == 0 and ast[0] != 1:
            res.append(ast[0] - 1)
        else:
            res.append(x)

        return res

File: 3227-find-missing-and-repeated-values/find-missing-and-repeated-values.js
/**
 * @param {number[][]} grid
 * @return {number[]}
 */
var findMissingAndRepeatedValues = function (grid) {
    let n = grid.length;
    let size = n * n;
    let count = new Array(size + 1).fill(0);

    for (let i = 0; i < n; i++) {
        for (let j = 0; j < n; j++) {
            count[grid[i][j]]++;
        }
    }

    let a = -1, b = -1;

    for (let num = 1; num <= size; num++) {
        if (count[num] === 2) {
            a = num;
        } else if (count[num] === 0) {
            b = num;
        }
    }

    return [a, b];
};